docs/advanced-topics/tools-vs-workflows.rst

```diff
@@ -0,0 +1,117 @@
+Tools vs Workflows: Detailed Comparison
+=======================================
+
+.. tip::
+  Generally speaking, if you care more about the Docker image than the descriptor file, and are using CWL, you are dealing with a tool. If you care more about the descriptor file, you are dealing with a workflow.
+
+History
+-------
+When Dockstore was created, CWL was the first descriptor language we supported. CWL has a very clear distinction between a tool and a workflow. However, our definition for each does not completely align with the language's specification -- strictly speaking, a CWL CommandLineTool does not require a Docker image, but we require tools to be associated with a Docker image on Dockstore. Furthermore, while other descriptor languages do not have separate concepts for tools and workflows, we still maintain a distinction between a tool and a workflow for WDL for legacy reasons.
+
+
+Should I write a tool or a workflow?
+------------------------------------
+.. hint::
+    We generally **discourage** people from writing/registering WDL tools and may deprecate them in the future.
+
+Dockstore tools are more associated with creating/owning Docker images that are used in conjunction with a descriptor language, and Dockstore workflows are more closely tied to the descriptor files themselves. If you control the Docker image being used by your CWL program, and want versioning to be based upon that image's tags, you should probably write a tool. If you do not control the Docker image being used, and/or are not writing in CWL, you should probably write a workflow. Of course, everyone's use case is different. We encourage you to look around Dockstore for inspiration for your next bioinformatics project.
+
+Overview
+--------
+
++------------------------+------------------------------------------+-------------------------------------------------+
+| Language               | Tool                                     | Workflow                                        |
++========================+==========================================+=================================================+
+| CWL                    |  - Class: CommandLineTool                |  - Class: Workflow                              |
+|                        |  - Dockerfile for the tool's image       |  - No Dockerfile required                       |
++------------------------+------------------------------------------+-------------------------------------------------+
+| WDL                    |  - One task that runs in a Docker image  |  - ≥1 task, which could run in Docker image     |
```

Comment:
> which could run in Docker image  

1. Technically they run in a container.
2. There can be more than one. Edit -- I see you cover this below, so maybe OK

Reply:
I made another version that replaces "image" with "container" where correct, but a side effects is that the user's browser window cannot be much smaller than 2370 px in width or else the table starts getting cut off. Then again, that's sort of implicit with most tables, I guess.

I wonder how it looks on mobile...

Reply:
>  wonder how it looks on mobile...

You can see with Developer tools in all browsers.

Reply:
How did I not know this?! This is incredible! 🤯 

Looks like the sidebar is hidden in mobile view, which is good. Also, it seems most devices turned landscape can at least see all three rows of the tables exist and would know to scroll to see the end of the last one. Portrait view, not so much, but I don't think supporting tables in portrait view on a non-tablet mobile device is high priority for us.

<img width="782" alt="Screen Shot 2022-01-31 at 7 39 59 PM" src="https://user-images.githubusercontent.com/27784612/151909580-515da5d4-699e-48f5-8b65-c734678bca55.png">